pennylane/_qubit_device.py

```diff
@@ -1649,8 +1649,10 @@ def marginal_prob(self, prob, wires=None):
         return self._reshape(prob, flat_shape)
 
     def expval(self, observable, shot_range=None, bin_size=None):
-        if observable.name == "Projector":
-            # branch specifically to handle the projector observable
+        if observable.name == "Projector" and len(observable.parameters[0]) == len(
+            observable.wires
+        ):
+            # branch specifically to handle the basis state projector observable
```

Comment:
What's going on here? Does `DefaultQubit.expval` have some tricks that are faster than the default?

Reply:
Here, the default qubit was branching off for the original projector class to speed things up exploiting the binary basis state to index the state probabilities. Now, we can still do this for `_BasisStateProjector`, but we can't pass `_StateVectorProjector` through this bit of code, which is instead treated as a regular observable. This is why we now not only check that the observable is a projector, but also check the shapes to ensure it is a `_BasisStateProjector`.